src/screens/AddressDetail.js

```diff
@@ -13,53 +13,53 @@ import hathorLib from '@hathor/wallet-lib';
 import ReactLoading from 'react-loading';
 import colors from '../index.scss';
 import WebSocketHandler from '../WebSocketHandler';
-import { TX_COUNT } from '../constants';
+import { TX_COUNT, TOKEN_COUNT } from '../constants';
 import { isEqual } from 'lodash';
-import helpers from '../utils/helpers';
 import metadataApi from '../api/metadataApi';
 import addressApi from '../api/addressApi';
+import txApi from '../api/txApi';
 
 
 class AddressDetail extends React.Component {
   pagination = new PaginationURL({
-    'hash': {required: false},
+    'token': {required: true},
     'page': {required: false},
-    'token': {required: true}
   });
 
   addressSummaryRef = React.createRef();
 
   /*
    * address {String} searched address (from url)
-   * balance {Object} Object with balance of each token of this address indexed by tokenUid {uid1: {'name', 'symbol', 'received', 'spent'}}
    * selectedToken {String} UID of the selected token when address has many
-   * numberOfTransactions {Number} Total number of transactions of the list
+   * balance {Object} Object with balance of the selected token on this address
    * transactions {Array} List of transactions history to show
-   * hasAfter {boolean} If has more elements after the list page
-   * hasBefore {boolean} If has more elements before the list page
    * queryParams {Object} Object with URL parameters data
    * loadingSummary {boolean} If is waiting response of data summary request
    * loadingHistory {boolean} If is waiting response of data history request
+   * loadingTokens {boolean} If is waiting response of tokens request
    * errorMessage {String} message to be shown in case of an error
    * warningRefreshPage {boolean} If should show a warning to refresh the page to see newest data for the address
+   *warnMissingTokens {number} If there are tokens that could not be fetched, this should be the total number of tokens.
    * selectedTokenMetadata {Object} Metadata of the selected token
    * metadataLoaded {boolean} When the selected token metadata was loaded
+   * addressTokens {Object} Object with all tokens that have passed on this address, indexed by token UID, i.e. {"00": {"name": "Hathor", "symbol": "HTR", "token_id": "00"}, ...}
    */
   state = {
     address: null,
-    balance: {},
     selectedToken: '',
-    numberOfTransactions: 0,
+    balance: {},
     transactions: [],
-    hasAfter: false,
-    hasBefore: false,
     queryParams: null,
-    loadingSummary: true,
+    loadingSummary: false,
     loadingHistory: false,
+    loadingTokens: true,
     errorMessage: '',
     warningRefreshPage: false,
+    warnMissingTokens: 0,
     selectedTokenMetadata: null,
     metadataLoaded: false,
+    addressTokens: {},
+    txCache: {},
```

Comment:
add `txCache` to the documentation above

Reply:
Done.